tests/net/lib/http_server/crime/src/main.c

```diff
@@ -4,85 +4,157 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#include <zephyr/ztest.h>
-#include <zephyr/net/http/client.h>
+#include <string.h>
 
-/*
- * The goals of CRiMe (Compressed Resources in Memory), are:
- *
- * 1. To support serving static http resources with minimal ROM footprint
- * 2. To do so with the least amount of fuss.
- *
- * For goal number 1, we know we can use python to read a file, compress the contents, and write
- * the compressed contents to another file. We can then wrap that python functionality in a cmake
- * wrapper.
- * 
- * For goal number 2, let's try and address the most common use cases listed below
- *
- * a) an html file
- * b) an html file and a client-side js file
- * c) a web root directory, automating as much content generation as possible
- * d) two web root directories, without shared resources
- * e) two web root directories, with shared resources
- *
- * Additionally, we would like to scale the python / cmake content generation from goal 1 to
- * happen for a number of resources, recursing into subdirectories, etc. At this time, there is no
- * requirement to support any kind of 'dot file' (configuration file), so all files can be treated
- * as simple static resources.
- *
- * The resources necessary to support the above uses cases could be something like what is shown
- * below:
- * 
- * ```
- * service-A
- * ├── css
- * │   └── service-A.css
- * ├── index.html
- * └── js
- *     └── service-A.js
- * service-B
- * ├── css
- * │   └── service-B.css
- * ├── index.html
- * └── js
- *     └── service-B.js
- * shared
- * └── 404.html
- * ```
- *
- * Testing each of these use cases will involve one or more successful HTTP GET requests, with
- * content validation for each request.
- */
+#include <zephyr/net/http/client.h>
+#include <zephyr/net/socket.h>
+#include <zephyr/ztest.h>
 
+static int sock_A;
+static int sock_B;
 static struct http_request req;
 static uint8_t buf[CONFIG_TEST_BUF_SIZE];
+static char port_str[sizeof("65535")];
+
+void http_client_req_create(const char *url, uint16_t port, http_response_cb_t cb)
+{
+	snprintf(port_str, sizeof(port_str), "%u", port);
+
+	req = (struct http_request){
+		.method = HTTP_GET,
+		.url = url,
+		.host = CONFIG_NET_CONFIG_MY_IPV4_ADDR,
+		.protocol = CONFIG_TEST_SERVER_PROTO,
+		.response = cb,
+		.recv_buf = buf,
+		.recv_buf_len = sizeof(buf),
+		.port = port_str,
+	};
+}
+
+static const struct http_service_desc *http_service_get_by_port(uint16_t port)
+{
+	const struct http_service_desc *svc;
+
+	HTTP_SERVICE_FOREACH(svc)
+	{
+		if (*svc->port == port) {
+			return svc;
+		}
+	}
+
+	return NULL;
+}
+
+static const struct http_service_desc *
+http_service_get_resource(const struct http_service_desc *scv, const char *resource)
+{
+	const struct http_resource_desc *res;
+	HTTP_SERVICE_FOREACH_RESOURCE(svc, res)
+	{
```

Comment:
ditto

Reply:
I think this is just the vs code auto formatter, but agred